base-math/floattitd.c

```diff
@@ -95,7 +120,7 @@ __BACKEND_ (floattitd) (INT128 a)
     {
       /* If abs(a) is less than 63-bits we can convert the signed
        * value directly.  */
-      result = (long long) a;
+    result = (INPUT_TYPE64) a;
```

Comment:
What happens when the return type (RET_TYPE?) is a Decimal32?
This is possible, right?